lib/std/secrets.nim

```diff
@@ -0,0 +1,186 @@
+## Cryptographically secure pseudorandom number generator.
+## 
+## | Targets    | Implementation|
+## | :---         | ----:       |
+## | Windows| `BCryptGenRandom`_ |
+## | Linux| `getrandom`_ system call when available, otherwise `/dev/urandom`_ will be used|
+## | Macosx| `SecRandomCopyBytes`_ system call when available, otherwise `/dev/urandom`_ will be used|
+## | OpenBSD| `getentropy`_ system call when available, otherwise `/dev/urandom`_ will be used|
+## | JS(Web Browser)| `getRandomValues`_|
+## | Other platforms| `/dev/urandom`_|
+##
+## .. _BCryptGenRandom: https://docs.microsoft.com/en-us/windows/win32/api/bcrypt/nf-bcrypt-bcryptgenrandom
+## .. _getrandom: https://man7.org/linux/man-pages/man2/getrandom.2.html
+## .. _/dev/urandom: https://en.wikipedia.org/wiki//dev/random
+## .. _SecRandomCopyBytes: https://developer.apple.com/documentation/security/1399291-secrandomcopybytes?language=objc
+## .. _getentropy: https://man.openbsd.org/getentropy.2
+## .. _getRandomValues: https://www.w3.org/TR/WebCryptoAPI/#Crypto-method-getRandomValues
+##
+
+import std/os
+
+runnableExamples:
+  doAssert urandom(0).len == 0
+  doAssert urandom(10).len == 10
+  doAssert urandom(20).len == 20
+  doAssert urandom(120).len == 120
+
+
+when defined(posix):
+  import std/posix
+
+  template processReadBytes(readBytes: int, p: pointer) =
+    if readBytes == 0:
+      break
+    elif readBytes > 0:
+      inc(result, readBytes)
+      cast[ptr pointer](p)[] = cast[pointer](cast[ByteAddress](p) + readBytes)
+    else:
+      if osLastError().int in {EINTR, EAGAIN}:
+        discard
+      else:
+        result = -1
+        break
+
+  proc getDevUrandom(p: pointer, size: int): int =
+    let fd = posix.open("/dev/urandom", O_RDONLY)
+
+    if fd > 0:
+      var stat: Stat
+      if fstat(fd, stat) != -1 and S_ISCHR(stat.st_mode):
+        while result < size:
+          let readBytes = posix.read(fd, p, size - result)
+          processReadBytes(readBytes, p)
+
+      discard posix.close(fd)
+
+when defined(js):
+  import std/private/jsutils
+
+  proc getRandomValues(arr: Uint8Array) {.importjs: "window.crypto.getRandomValues(#)".}
+
+  const
+    maxBrowserCryptoBufferSize = 256
+
+  proc urandom*(p: var openArray[byte]): int =
+    let size = p.len
+    if size > 0:
+      let chunks = (size - 1) div maxBrowserCryptoBufferSize
```

Comment:
I also find `chunks` apis are useful:
https://doc.rust-lang.org/std/slice/struct.Chunks.html
https://doc.rust-lang.org/std/slice/struct.ChunksMut.html

Reply:
std/views (from https://github.com/nim-lang/Nim/pull/14869 which was wrongly closed) would enable things and many other things